### PR TITLE
Link to more existing examples from new examples page

### PR DESCRIPTION
**Pull request by crusso**
Fixes #637 

I guess it would be wise to update the examples repo to 0.8.4 - see PR https://github.com/dfinity/examples/pull/129.

### CODE REVIEW

File: modules/examples/pages/index.adoc [hunk 1]
```diff
@@ -10,31 +10,63 @@ ifdef::env-github,env-browser[:outfilesuffix:.adoc]
...
-- https://github.com/dfinity/examples/tree/master/motoko/life[Game of Life] - Conway's https://en.wikipedia.org/wiki/Conway%27s_Game_of_Life[Game of Life], running in a Motoko Canister
+- https://github.com/dfinity/examples/tree/master/motoko/classes[classes] -- dynamic actor (class) instantiation.
+- https://github.com/dfinity/examples/tree/master/motoko/pub-sub[pub-sub]-- multiple canisters, with publisher-subscriber inter-canister calls.
+- https://github.com/dfinity/examples/tree/master/motoko/quicksort[quicksort] -- sorting an array, via Quick Sort, in Motoko.
+- https://github.com/dfinity/examples/tree/master/motoko/simple-to-do[simple-to-do] -- CRUD-like demo service, sans a front end; see also: phone-book and superheroes.
+- https://github.com/dfinity/examples/tree/master/motoko/calc[calc] -- more advanced version of counter demo.
+
+=== Minimal front end.
+
+These examples use a minimal front end component.
```
Comment (emmaperetti) [inline modules/examples/pages/index.adoc:40]:
```suggestion
These examples use a minimal front end component. 
```

File: modules/examples/pages/index.adoc [hunk 2]
```diff
@@ -10,31 +10,63 @@ ifdef::env-github,env-browser[:outfilesuffix:.adoc]
...
-- https://github.com/dfinity/examples/tree/master/motoko/superheroes[Superheroes] - a simple CRUD dapp
-- https://github.com/dfinity/examples/tree/master/motoko/hello_cycles[Hello Cycles] - Receive and transfer cycles and check your balance
-- https://m7sm4-2iaaa-aaaab-qabra-cai.raw.ic0.app/[Motoko Playground] - Play with other Motoko examples on the playground
+- https://github.com/dfinity/examples/tree/master/motoko/actor_reference[actor_reference] -- IC management canister as an actor (reference).
+- https://github.com/dfinity/examples/tree/master/motoko/counter[counter] -- basic (stable) variable demo.
+- https://github.com/dfinity/examples/tree/master/motoko/echo[echo] -- basic query function demo.
+- https://github.com/dfinity/examples/tree/master/motoko/factorial[factorial] -- basic factorial demo.
+- https://github.com/dfinity/examples/tree/master/motoko/hello-world[hello-world] -- basic hello world demo.
+- https://github.com/dfinity/examples/tree/master/motoko/hello_cycles[hello_cycles] -- basic cycle transfer demo.
+- https://github.com/dfinity/examples/tree/master/motoko/whoami[whoami] -- basic caller identification demo.
```
Comment (emmaperetti) [inline modules/examples/pages/index.adoc:28]:
```suggestion
- https://github.com/dfinity/examples/tree/master/motoko/whoami[whoami] -- a simple canister that reports the Principal of its caller.
```

File: modules/examples/pages/index.adoc [hunk 3]
```diff
@@ -10,31 +10,63 @@ ifdef::env-github,env-browser[:outfilesuffix:.adoc]
...
-- https://github.com/dfinity/examples/tree/master/motoko/whoami[Whoami] - a simple canister that reports the Principal of its caller
-- https://github.com/dfinity/examples/tree/master/motoko/superheroes[Superheroes] - a simple CRUD dapp
-- https://github.com/dfinity/examples/tree/master/motoko/hello_cycles[Hello Cycles] - Receive and transfer cycles and check your balance
-- https://m7sm4-2iaaa-aaaab-qabra-cai.raw.ic0.app/[Motoko Playground] - Play with other Motoko examples on the playground
+- https://github.com/dfinity/examples/tree/master/motoko/actor_reference[actor_reference] -- IC management canister as an actor (reference).
+- https://github.com/dfinity/examples/tree/master/motoko/counter[counter] -- basic (stable) variable demo.
+- https://github.com/dfinity/examples/tree/master/motoko/echo[echo] -- basic query function demo.
+- https://github.com/dfinity/examples/tree/master/motoko/factorial[factorial] -- basic factorial demo.
+- https://github.com/dfinity/examples/tree/master/motoko/hello-world[hello-world] -- basic hello world demo.
+- https://github.com/dfinity/examples/tree/master/motoko/hello_cycles[hello_cycles] -- basic cycle transfer demo.
```
Comment (emmaperetti) [inline modules/examples/pages/index.adoc:27]:
```suggestion
- https://github.com/dfinity/examples/tree/master/motoko/hello_cycles[hello_cycles] -- Receive and transfer cycles and check your balance.
```

File: modules/examples/pages/index.adoc [hunk 4]
```diff
@@ -10,31 +10,63 @@ ifdef::env-github,env-browser[:outfilesuffix:.adoc]
...
+These examples use a minimal front end component.
+
+- https://github.com/dfinity/examples/tree/master/motoko/random_maze[random_maze] -- random maze generation, with IC-based randomness.
+- https://github.com/dfinity/examples/tree/master/motoko/cert-var[cert_var] -- simple certified variable (a single 32-bit number), with client-side certificate validation.
+
+=== Full stack.
+
+These examples use a "conventional" front end component (via `React.Component`).
+
+- https://github.com/dfinity/examples/tree/master/motoko/life[life] -- demonstrates upgrades among three versions and stable state migration.
```
Comment (emmaperetti) [inline modules/examples/pages/index.adoc:49]:
```suggestion
- https://github.com/dfinity/examples/tree/master/motoko/life[life] -- Conway's https://en.wikipedia.org/wiki/Conway%27s_Game_of_Life[Game of Life], running in a Motoko Canister. Demonstrates upgrades among three versions and stable state migration. 
```

**Review by emmaperetti — APPROVED** [review]

**Comment by crusso** [comment]
@krpeacock @emmaperetti and I took a pass adding back more of the examples. PTAL

File: modules/examples/pages/index.adoc [hunk 5]
```diff
@@ -10,31 +10,63 @@ ifdef::env-github,env-browser[:outfilesuffix:.adoc]
 In addition to our tutorials section, here are some quick references! You can also check out our https://github.com/dfinity/examples[Examples Repo] or https://github.com/dfinity/awesome-dfinity[Awesome Dfinity], a curated set of open-source projects and examples from our community!
 
 [[motoko]]
-== Motoko Examples
+== Motoko
```
Comment (crusso) [inline modules/examples/pages/index.adoc:13]:
```suggestion
== Motoko Examples
```

**Review by emmaperetti — APPROVED** [review]

File: modules/examples/pages/index.adoc [hunk 6]
```diff
@@ -12,29 +12,61 @@ In addition to our tutorials section, here are some quick references! You can al
...
 
 [[motoko-advanced]]
 === Advanced Examples
 
+Large examples contributed by the community and not necessarily up-to-date:
+
+- Reversi game, in Motoko https://github.com/ninegua/reversi[src] accompanying https://ninegua.github.io/reversi[Blog post]
+
+- QR code generator in Motoko https://github.com/enzoh/motoko-qr[src] accompanying
+  https://medium.com/@ehaussecker/my-first-microservice-on-dfinity-3ac5c142865bB[Medium article]
```
Comment (crusso) [inline modules/examples/pages/index.adoc:63]: https://medium.com/@ehaussecker/my-first-microservice-on-dfinity-3ac5c142865b
```suggestion
  https://medium.com/@ehaussecker/my-first-microservice-on-dfinity-3ac5c142865b[Medium article]
```

File: modules/examples/pages/index.adoc [hunk 7]
```diff
@@ -51,5 +83,15 @@ First, here are some basic examples for some introductory use-cases:
 - https://www.youtube.com/watch?v=36L33S_DYHY&ab_channel=DFINITY[Community Conversations | Best Practices for Canisters in Rust]
 
 
-
 More Rust examples coming soon!
+
```
Comment (chenyan-dfinity) [inline modules/examples/pages/index.adoc:87]: There is also a counter example in wat.

Comment (crusso) [reply modules/examples/pages/index.adoc:87]: Ok, I'll reference that.

File: modules/examples/pages/index.adoc [hunk 8]
```diff
@@ -51,5 +83,15 @@ First, here are some basic examples for some introductory use-cases:
...
+== C
+
+For examples of projects written in the C programming language for the {IC}, see link:https://github.com/dfinity/examples/tree/master/c[C++ sample projects].
+
+WARNING: these examples are likely to bit-rot and may need updating and are provided for illustration only.
+
+* link:https://github.com/dfinity/examples/tree/master/c/adventure[Adventure game]
+* link:https://github.com/dfinity/examples/tree/master/c/qr[QR code generator]
+* link:https://github.com/dfinity/examples/tree/master/c/reverse[Reverse game]
+
```
Comment (crusso) [inline modules/examples/pages/index.adoc:97]:
```suggestion

== Wat (Wasm textual format)

Just for fun, we also have a simple example written in pure Wasm textual format:

* link:https://github.com/dfinity/examples/tree/master/wasm/counter[counter]

```

**Comment by crusso** [comment]
@chenyan-dfinity PTAL

File: modules/examples/pages/index.adoc [hunk 9]
```diff
@@ -12,29 +12,61 @@ In addition to our tutorials section, here are some quick references! You can al
...
 === Advanced Examples
 
+Large examples contributed by the community and not necessarily up-to-date:
+
+- Reversi game, in Motoko https://github.com/ninegua/reversi[src] accompanying https://ninegua.github.io/reversi[Blog post]
+
+- QR code generator in Motoko https://github.com/enzoh/motoko-qr[src] accompanying
+  https://medium.com/@ehaussecker/my-first-microservice-on-dfinity-3ac5c142865b[Medium article]
+
 - https://github.com/timohanke/icvault[IC Vault] - End-to-end Encryption
```
Comment (crusso) [inline modules/examples/pages/index.adoc:65]:
```suggestion
- https://github.com/timohanke/icvault[IC Vault] - end-to-end Encryption.
```

File: modules/examples/pages/index.adoc [hunk 10]
```diff
@@ -12,29 +12,61 @@ In addition to our tutorials section, here are some quick references! You can al
...
+- https://github.com/dfinity/examples/tree/master/motoko/cert-var[cert_var] -- simple certified variable (a single 32-bit number), with client-side certificate validation.
+
+=== Full stack.
+
+These examples use a "conventional" front end component (via `React.Component`).
+
+- https://github.com/dfinity/examples/tree/master/motoko/life[life] -- Conway's https://en.wikipedia.org/wiki/Conway%27s_Game_of_Life[Game of Life], running in a Motoko Canister. Demonstrates upgrades among three versions and state migration using stable variables. 
+- https://github.com/dfinity/examples/tree/master/motoko/phone-book[phone-book] -- CRUD-like demo service.
+- https://github.com/dfinity/examples/tree/master/motoko/superheroes[superheroes] -- a simple CRUD dapp. 
 - https://github.com/krpeacock/auth-client-demo[Auth Client Demo] - Simple authentication with Internet Identity
```
Comment (crusso) [inline modules/examples/pages/index.adoc:52]:
```suggestion
- https://github.com/krpeacock/auth-client-demo[Auth Client Demo] - simple authentication with Internet Identity.
```

File: modules/examples/pages/index.adoc [hunk 11]
```diff
@@ -12,29 +12,61 @@ In addition to our tutorials section, here are some quick references! You can al
...
 - https://github.com/krpeacock/ic-avatar[IC-Avatar] - Full-stack Motoko + React with more advanced authentication, image uploads, and optimistic updates
```
Comment (crusso) [inline modules/examples/pages/index.adoc:53]:
```suggestion
- https://github.com/krpeacock/ic-avatar[IC-Avatar] - full-stack Motoko + React with more advanced authentication, image uploads, and optimistic updates.
```

File: modules/examples/pages/index.adoc [hunk 12]
```diff
@@ -12,29 +12,61 @@ In addition to our tutorials section, here are some quick references! You can al
...
 
 [[motoko-advanced]]
 === Advanced Examples
 
+Large examples contributed by the community and not necessarily up-to-date:
+
+- Reversi game, in Motoko https://github.com/ninegua/reversi[src] accompanying https://ninegua.github.io/reversi[Blog post]
```
Comment (crusso) [inline modules/examples/pages/index.adoc:60]:
```suggestion
- Reversi game, in Motoko https://github.com/ninegua/reversi[src] accompanying https://ninegua.github.io/reversi[Blog post].
```

File: modules/examples/pages/index.adoc [hunk 13]
```diff
@@ -12,29 +12,61 @@ In addition to our tutorials section, here are some quick references! You can al
...
+
+- QR code generator in Motoko https://github.com/enzoh/motoko-qr[src] accompanying
+  https://medium.com/@ehaussecker/my-first-microservice-on-dfinity-3ac5c142865b[Medium article]
```
Comment (crusso) [inline modules/examples/pages/index.adoc:63]:
```suggestion
  https://medium.com/@ehaussecker/my-first-microservice-on-dfinity-3ac5c142865b[Medium article].
```

**Review by chenyan-dfinity — APPROVED** [review]

**Pull request merged**